libs/blocks/preflight/panels/performance.js

```diff
@@ -0,0 +1,277 @@
+    failsWhen: !validLcp,
+    key: text.lcpEl.key,
+    icon: icons.pass,
+    description: text.lcpEl.passed.description,
+  });
+  return validLcp;
+}
+
+export const checkFragments = () => conditionalItemUpdate({
+  failsWhen: config.lcp.element.closest('.fragment') || config.lcp.element.closest('.section')?.querySelector('[data-path*="fragment"]'),
+  key: text.fragments.key,
+});
+
+export const checkPlaceholders = async () => conditionalItemUpdate({
+  failsWhen: config.lcp.element.closest('.section').dataset.placeholdersDecorated === 'true',
+  key: text.placeholders.key,
+});
+
+export const checkForPersonalization = async () => conditionalItemUpdate({
+  failsWhen: getMetadata('personalization') || getMetadata('target') === 'on',
+  key: text.personalization.key,
+});
+
+export const checkVideosWithoutPosterAttribute = async () => conditionalItemUpdate({
+  failsWhen: !config.lcp.element.poster,
+  emptyWhen: !config.lcp.url.match('media_.*.mp4'),
+  key: text.videoPoster.key,
+});
+
+export const checkIcons = () => conditionalItemUpdate({
+  failsWhen: config.lcp.element.closest('.section').querySelector('.icon-milo'),
+  key: text.icons.key,
+});
+
+export function PerformanceItem({ icon, title, description }) {
+  return html` <div class="seo-item">
+    <div class="result-icon ${icon}"></div>
+    <div class="seo-item-text">
+      <p class="seo-item-title">${title}</p>
+      <p class="seo-item-description">${description}</p>
+    </div>
+  </div>`;
+}
+
+export const checkForSingleBlock = () => conditionalItemUpdate({
+  failsWhen: document.querySelector('main > div.section').childElementCount > 1,
+  key: text.singleBlock.key,
+});
+
+export const createPerformanceItem = ({
+  icon,
+  title,
+  description,
+} = {}) => html`<${PerformanceItem}
+  icon=${icon}
+  title=${title}
+  description=${description}
+/>`;
+
+let clonedLcpSection;
+function highlightElement(event) {
+  if (!config.lcp) return;
+  const lcpSection = config.lcp?.element.closest('.section');
+  const tooltip = document.querySelector('.lcp-tooltip-modal');
+  const { offsetHeight, offsetWidth } = lcpSection;
+  const scaleFactor = Math.min(500 / offsetWidth, 500 / offsetHeight);
+  if (!clonedLcpSection) {
+    clonedLcpSection = lcpSection.cloneNode(true);
+    clonedLcpSection.classList.add('lcp-clone');
+    clonedLcpSection.style.width = `${lcpSection.offsetWidth}px`;
+    clonedLcpSection.style.height = `${lcpSection.offsetHeight}px`;
+    clonedLcpSection.style.transform = `scale(${scaleFactor})`;
+    clonedLcpSection.style.transformOrigin = 'top left';
+  }
+  if (!tooltip.children.length) tooltip.appendChild(clonedLcpSection);
+  const { top, left } = event.currentTarget.getBoundingClientRect();
+  tooltip.style.width = `${offsetWidth * scaleFactor}px`;
+  tooltip.style.height = `${offsetHeight * scaleFactor}px`;
+  tooltip.style.top = `${top + window.scrollY - offsetHeight * scaleFactor - 10}px`;
+  tooltip.style.left = `${left + window.scrollX}px`;
```

Comment:
I will echo Narcis and point out that we can use `Object.assign` in here too.
```suggestion
Object.assign(tooltip.style, {
  width: `${offsetWidth * scaleFactor}px`,
  height: `${offsetHeight * scaleFactor}px`,
  top: `${top + window.scrollY - offsetHeight * scaleFactor - 10}px`,
  left: `${left + window.scrollX}px`
});
```